src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp

```diff
@@ -88,17 +89,41 @@ void handle_reshape::run(program& p) {
             if (input_node.is_type<reorder>())
                 continue;
 
-            node->get_output_layout();
+            auto output_layout = node->get_output_layout();
 
             // vector for storing nodes that are reorder type, for which splitted primitives are needed (except for the
             // first one where orginal reshape will be used)
             std::vector<program_node*> reorder_node_to_split;
             std::vector<program_node*> onednn_users;
 
+            // In case reshape is used as an user of loop operation and the type is integer,
+            // the output of reshape node needs not to change from ngraph.
+            bool need_reorder_node_to_split = true;
+            if (output_layout.data_type == ov::element::Type_t::i32 ||
+                output_layout.data_type == ov::element::Type_t::i64 ||
+                output_layout.data_type == ov::element::Type_t::u32 ||
+                output_layout.data_type == ov::element::Type_t::u64) {
+                auto node_id = node->id();
+                for (auto& user : node->get_users()) {
+                    program_node& loop_node = *user;
+                    if (loop_node.is_type<loop>()) {
+                        auto& loop_prim = loop_node.as<loop>();
+                        if (loop_prim.get_trip_count_id().compare(node_id) == 0 ||
+                            loop_prim.get_initial_execution_id().compare(node_id) == 0 ||
+                            loop_prim.get_current_iteration_id().compare(node_id) == 0 ||
+                            loop_prim.get_execution_condition_id().compare(node_id) == 0 ||
+                            loop_prim.get_num_iterations_id().compare(node_id) == 0) {
+                            need_reorder_node_to_split = false;
+                            break;
+                        }
+                    }
+                }
+            }
```

Comment:
Based on your explanation from the meeting, I think this is not about loop, but general issue. Could you fix the real problem, not the symptom-on-loop?

Reply:
Yes, it's updated with general fix for reshape and reorder.